core/src/avm2/e4x.rs

```diff
@@ -1055,14 +1055,18 @@ impl<'gc> E4XNode<'gc> {
             return false;
         }
 
+        // The Multiname is not a QName, so an any name matches everything.
+        // See https://github.com/adobe/avmplus/blob/858d034a3bd3a54d9b70909386435cf4aec81d21/core/Multiname.cpp#L59
+        if name.is_any_name() && !name.is_qname() {
+            return true;
+        }
+
         if !name.is_any_name() && self.local_name() != name.local_name() {
             return false;
         }
 
-        // The Multiname is not a QName, so an any name matches everything.
-        // See https://github.com/adobe/avmplus/blob/858d034a3bd3a54d9b70909386435cf4aec81d21/core/Multiname.cpp#L59
-        if name.is_any_name() && name.namespace_set().len() > 1 {
-            return true;
+        if self.local_name().is_none() {
```

Comment:
It would be good to add a comment that explains what this does. I think in essence we/Flash allow matching non-elements nodes like text (that don't have a name) in specific instances, i.e. not a qname?

I would also move this above the previous if.

Reply:
FWIW, the logic after this PR directly line-per-line corresponds to equivalent logic in avmplus:

https://github.com/adobe/avmplus/blob/858d034a3bd3a54d9b70909386435cf4aec81d21/core/Multiname.cpp#L59-L71

So IMO unless we have a really good reason, I don't think the check order should be changed here.

The comments I can expand if needed.